R/textmodel_nb.R

```diff
@@ -115,7 +115,9 @@ textmodel_nb.dfm <- function(x, y, smooth = 1,
     distribution <- match.arg(distribution)
     call <- match.call()
     
-    y <- factor(y)
+    y <- as.factor(y)
+    if (length(unique(na.omit(y))) == 1) stop("y cannot be constant")
```

Comment:
How about `var(as.numeric(y), na.rm = TRUE)) == 0`?

Reply:
OK, that's most efficient (although the differences are tiny). As long as we don't get into any numerical precision issues, then we should be ok.
```r
> x <- as.factor(sample(letters[1:3], size = 1000000, replace = TRUE))
> microbenchmark(unique = (length(unique(na.omit(x))) == 1), 
+                var = var(as.numeric(x), na.rm = TRUE) == 0,
+                dupl = all(duplicated(x)[-1L]))
Unit: milliseconds
   expr       min        lq     mean    median       uq       max neval
 unique 15.457192 17.864760 22.85690 19.102567 22.57854  66.27518   100
    var  6.526918  7.920738 14.83654  9.182764 15.35607 221.80587   100
   dupl 13.403758 16.811788 23.87359 18.691379 25.27883 227.15093   100
```